### PR TITLE
[WIP] Make ticket layout configurable

### PR DESCRIPTION
**Pull request by siddhantbajaj**
Admin can set the layout type for the tickets of the conference.

screen shot 2017-07-06 at 2 15 42 pm

### CODE REVIEW

**Comment by Ana06** [comment]
[Make ticket layout configurable](https://trello.com/c/Y5A4kjMq/13-make-ticket-layout-configurable)

File: app/views/admin/conferences/edit.html.haml [hunk 1]
```diff
@@ -11,6 +11,7 @@
       = f.input :short_title, hint: "A short title, e.g. 'oSC14', to be used in URLs"
       = f.input :description, hint: markdown_hint('A description of the conference.'), input_html: { rows: 5, data: { provide: 'markdown-editable' } }
       = f.input :color, hint: 'The color will be used eg for the dashboard.', input_html: {size: 6, type: 'color'}
+      = f.input :ticket_layout, as: :select, collection: Conference.ticket_layouts.keys, hint: "Layout type for tickets of the conference."
```
Comment (Ana06) [inline app/views/admin/conferences/edit.html.haml]: I would exchange the ticket layout and the conference picture, I mean moving ticket layout down as the conference picture is more important :wink:

File: app/models/conference.rb [hunk 2]
```diff
@@ -73,6 +74,11 @@ class Conference < ActiveRecord::Base
   after_create :create_free_ticket
   after_update :delete_event_schedules
 
+  enum ticket_layout: {
```
Comment (Ana06) [inline app/models/conference.rb]: Why about writing this as array: 

http://edgeguides.rubyonrails.org/4_1_release_notes.html#active-record-enums

File: app/views/admin/conferences/edit.html.haml (hunk 1 above)
Comment (Ana06) [inline app/views/admin/conferences/edit.html.haml]: `collection` would be written as `Conference.ticket_layouts` without the `keys` if you use and array :wink:

File: app/views/physical_ticket/show.pdf.prawn [hunk 3]
```diff
@@ -1,4 +1,4 @@
-prawn_document(filename: @file_name, page_layout: :portrait, :page_size =>'A4' ) do |pdf|
+prawn_document(filename: @file_name, page_layout: @conference.ticket_layout.to_sym, :page_size =>'A4' ) do |pdf|
```
Comment (Ana06) [inline app/views/physical_ticket/show.pdf.prawn]: maybe moving `@conference.ticket_layout.to_sym` to the controller

This way you can also check that this variable is correctly assigned in the tests. :wink:

**Comment by Ana06** [comment]
And the tests are failing :fearful:

File: app/views/admin/conferences/edit.html.haml (hunk 1 above)
Comment (siddhantbajaj) [inline app/views/admin/conferences/edit.html.haml]: But `Conference.ticket_layouts` would still return a map even if we use array. So I think we would still need `Conference.ticket_layouts.keys` as `collection`, right?

**Comment by siddhantbajaj** [comment]
Updated Paper Trail. This [Issue](https://github.com/airblade/paper_trail/issues/798) was causing test failure.

File: app/views/admin/conferences/edit.html.haml (hunk 1 above)
Comment (mdeniz) [inline app/views/admin/conferences/edit.html.haml]: Looks likes yes, it returns a Hash

Comment (Ana06) [inline app/views/admin/conferences/edit.html.haml]: yes, sorry :confounded:

**Pull request merged**